pdns/dnsdist-lua.cc

```diff
@@ -379,134 +357,160 @@ static void setupLuaConfig(LuaContext& luaCtx, bool client, bool configCheck)
-      }
-
-      if(vars.count("checkFunction")) {
-        ret->checkFunction= boost::get<DownstreamState::checkfunc_t>(vars["checkFunction"]);
+      if (getOptionalValue<std::string>(vars, "maxInFlight", valueStr) > 0){
+        try {
+          ret->d_maxInFlightQueriesPerConn = boost::lexical_cast<size_t>(valueStr);
+        } catch (std::exception& e) {
+          warnlog("maxInFlight must be non-negative integer, not '%s' - ignoring", valueStr);
+        }
       }
 
-      if(vars.count("checkTimeout")) {
-        ret->checkTimeout = std::stoi(boost::get<string>(vars["checkTimeout"]));
+      if (getOptionalValue<std::string>(vars, "name", valueStr) > 0) {
+        ret->setName(valueStr);
       }
 
-      if(vars.count("setCD")) {
-        ret->setCD=boost::get<bool>(vars["setCD"]);
+      if (getOptionalValue<std::string>(vars, "id", valueStr) > 0) {
+        try {
+          ret->setId(boost::lexical_cast<boost::uuids::uuid>(valueStr));
+        } catch (std::exception& e) {
+          errlog("Error creating new server: id must be a valid UUID, not '%s'", valueStr);
+          return ret;
+        }
       }
 
-      if(vars.count("mustResolve")) {
-        ret->mustResolve=boost::get<bool>(vars["mustResolve"]);
+      if (getOptionalValue<std::string>(vars, "checkName", valueStr) > 0) {
+        ret->checkName = DNSName{valueStr};
       }
+      getOptionalValue<std::string>(vars, "checkType", ret->checkType);
 
-      if(vars.count("useClientSubnet")) {
-        ret->useECS=boost::get<bool>(vars["useClientSubnet"]);
+      if (getOptionalValue<std::string>(vars, "checkClass", valueStr) > 0) {
+        try {
+          ret->checkClass = boost::lexical_cast<int>(valueStr);
+        } catch (std::exception& e) {
+          warnlog("checkClass must be integer, not '%s' - ignoring", valueStr);
+        }
       }
 
-      if(vars.count("useProxyProtocol")) {
-        ret->useProxyProtocol = boost::get<bool>(vars["useProxyProtocol"]);
-      }
+      getOptionalValue<DownstreamState::checkfunc_t>(vars, "checkFunction",  ret->checkFunction);
 
-      if(vars.count("disableZeroScope")) {
-        ret->disableZeroScope=boost::get<bool>(vars["disableZeroScope"]);
+      if (getOptionalValue<std::string>(vars, "checkTimeout", valueStr) > 0) {
+        try {
+          ret->checkTimeout = boost::lexical_cast<int>(valueStr);
+        } catch (std::exception& e) {
+          warnlog("checkTimeout must be integer, not '%s' - ignoring", valueStr);
+        }
       }
 
-      if(vars.count("ipBindAddrNoPort")) {
-        ret->ipBindAddrNoPort=boost::get<bool>(vars["ipBindAddrNoPort"]);
-      }
+      getOptionalValue<bool>(vars, "setCD", ret->setCD);
+      getOptionalValue<bool>(vars, "mustResolve", ret->mustResolve);
+      getOptionalValue<bool>(vars, "useClientSubnet", ret->useECS);
+      getOptionalValue<bool>(vars, "useProxyProtocol", ret->useProxyProtocol);
+      getOptionalValue<bool>(vars, "disableZeroScoping", ret->disableZeroScope);
```

Comment:
```suggestion
      getOptionalValue<bool>(vars, "disableZeroScope", ret->disableZeroScope);
```

Reply:
hm. that would also change the parameter name no?

Reply:
I believe the name is currently `disableZeroScope`?